lib/mp4/transmuxer.js

```diff
@@ -49,29 +66,61 @@ AudioSegmentStream = function(track) {
 
     // buffer audio data until end() is called
     aacFrames.push(data);
-    aacFramesLength += data.data.byteLength;
   };
 
-  this.setEarliestDts = function (earliestDts) {
+  this.setEarliestDts = function(earliestDts) {
     earliestAllowedDts = earliestDts - track.timelineStartInfo.baseMediaDecodeTime;
   };
 
   this.flush = function() {
-    var boxes, currentFrame, data, sample, i, mdat, moof;
+    var
+      frames,
+      moof,
+      mdat,
+      boxes;
+
     // return early if no audio data has been observed
-    if (aacFramesLength === 0) {
+    if (aacFrames.length === 0) {
       this.trigger('done');
       return;
     }
 
-    // If the audio segment extends before the earliest allowed dts
-    // value, remove AAC frames until starts at or after the earliest
-    // allowed dts.
+    frames = this.trimAacFramesByEarliestDts_(aacFrames);
+
+    // we have to build the index from byte locations to
+    // samples (that is, aac frames) in the audio data
+    track.samples = this.generateSampleTable_(frames);
+
+    // concatenate the audio data to constuct the mdat
+    mdat = mp4.mdat(this.concatenateFrameData_(frames));
+
+    aacFrames = [];
+
+    calculateTrackBaseMediaDecodeTime(track);
+    moof = mp4.moof(sequenceNumber, [track]);
+    boxes = new Uint8Array(moof.byteLength + mdat.byteLength);
+
+    // bump the sequence number for next time
+    sequenceNumber++;
+
+    boxes.set(moof);
+    boxes.set(mdat, moof.byteLength);
+
+    clearDtsInfo(track);
+
+    this.trigger('data', {track: track, boxes: boxes});
+    this.trigger('done');
+  };
+
+  // If the audio segment extends before the earliest allowed dts
+  // value, remove AAC frames until starts at or after the earliest
+  // allowed DTS
+  this.trimAacFramesByEarliestDts_ = function(aacFrames) {
     if (track.minSegmentDts < earliestAllowedDts) {
```

Comment:
I would consider making `earliestAllowedDts` an argument to this function to make the dependencies clearer now that it's headed towards a general purpose utility. Not worth holding up the PR for, though.